app/controllers/solidus_subscriptions/api/v1/subscriptions_controller.rb

```diff
@@ -6,6 +6,26 @@ module V1
       class SubscriptionsController < BaseController
         protect_from_forgery unless: -> { request.format.json? }
 
+        def create
+          store = params[:store_id].nil? ? ::Spree::Store.default : ::Spree::Store.find(id: params[:store_id])
+          attributes = create_subscription_params.merge(user: current_api_user, store: store)
+          payment_source_type = attributes[:payment_source_type]
+
+          if payment_source_type.blank? || valid_payment_source_type?(payment_source_type)
```

Comment:
I'm not sure I like the idea of passing the payment source type directly, because it's an implementation detail rather than something the API consumers should be concerned with.

Could we infer it from the payment method instead? `Spree::PaymentMethod` has an abstract `payment_source_class` method that we could use.

Reply:
Good point, I agree.
What about this changes? [`5a47510` (#199)](https://github.com/solidusio-contrib/solidus_subscriptions/pull/199/commits/5a47510983040ffc0c5889f289632ab408808314)